docs/ARCHITECTURE.md

```diff
@@ -201,3 +201,67 @@ can be found under various directories such as `factory/`, `modprobe.d/`, `netwo
 `tools/`, `coccinelle/`, `.github/`, `.semaphore/`, `.mkosi/` host various
 utilities and scripts that are used by maintainers and developers. They are not
 shipped or installed.
+
+## UEFI
+The systemd project provides a UEFI bootloader, `systemd-boot`, and a stub that can be attached to a kernel
+image to allow loading it as a UEFI executable, `systemd-stub`. These components fully support UEFI Secure
+Boot, and this paragraph will describe their security posture and how they comply with industry-standard
+expectations for UEFI Secure Boot workflows.
+
+### Core Design
+The core design goals for these components is simplicity, achieved by offloading security-critical tasks to
+the firmware or earlier stages of the boot process (e.g.: `Shim`).
+
+When Secure Boot is enabled, these components are designed to avoid executing or loading or using
+unauthenticated payloads that could compromise the boot process, with special care taken for anything that
+could affect the system before `ExitBootServices()` has been called. For example, when additional resources
+are loaded, if running with Secure Boot enabled, they will be validated before use. The only exceptions are
+the bootloader's own textual configuration files. There are no build time or runtime configuration options
+that can be set to weaken the security model of these components when Secure Boot is enabled.
```

Comment:
So here's a thing to mention: sd-boot will populate its menu from data extracted from untrusted data (specifically: type1 drop-ins and type2 PE info extracted *before* authenticating them). It uses this for display purposes, and ordering entries. When a boot menu item is invoked it is authenticated fully however, and sd-stub ensures it won't take any parameters from untrusted soures.

Reply:
mmh I didn't realize we parsed PEs before validating them, perhaps that's something to fix, but it's for another day